go/marshal/encode.go

```diff
@@ -111,6 +112,42 @@ func MustMarshal(v interface{}) types.Value {
 	return r
 }
 
+// MarshalOptions customize how a type is marshalled.
+//
+// For example, to specify a name for a struct when it's marshalled:
+//
+//  type MyStruct struct {
+//    Foo string
+//  }
+//
+//  func (ms MyStruct) GetMarshalOptions() (opts marshal.MarshalOptions) {
+//    opts.Name = "SomeOtherName"
```

Comment:
Yeah I know you can do that. The pattern I said:
```go
opts.Name = "SomeOtherName"
```
is nice because it implies not changing other options (especially if you want to somehow use a placeholder value for the name). Also you don't need to type `marshal.MarshalOptions` multiple times :-)